GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+  Deposit       uint64  // Number of Atoms to add to the proposal's deposit
+}
+
+```
+
+A `TxDeposit` transaction has to go through a number of checks to be valid. These checks are outlined in the following pseudocode.
+
+```
+// PSEUDOCODE //
+// Check if TxDeposit is valid. If it is, increase deposit and check if MinDeposit is reached
+
+upon receiving txDeposit from sender do
+  // check if proposal is correctly formatted. Includes fee payment.
+  
+  if !correctlyFormatted(txDeposit) then  
+    throw
+  
+  else
+    if !exist(txDeposit.proposalID) then  
+      // There is no proposal for this proposalID
+      
+      throw
+    
+    else
+      if (txDeposit.Deposit <= 0 OR sender.AtomBalance < txDeposit.Deposit)
+        // deposit is negative or null OR sender has insufficient funds
+        
+        throw
+      
+      else
+        retrieve proposal from txDeposit.ProposalID // retrieve throws if it fails
+        
+        if (proposal.Deposit >= ActiveProcedure.MinDeposit) then  
+          // MinDeposit was reached
+          
+          throw
+        
+        else
+          if (CurrentBlock >= proposal.SubmitBlock + ActiveProcedure.MaxDepositPeriod) then 
+            // Maximum deposit period reached
+            
+            throw
+          
+          else
+            // sender can deposit
+            
+            retrieve depositorsList from txDeposit.ProposalID
+            sender.AtomBalance -= txDeposit.Deposit
+
+            if sender is in depositorsList
+              increase deposit of sender in depositorsList by txDeposit.Deposit
+
+            else
+              initialise deposit of sender in depositorsList at txDeposit.Deposit
+            
+            proposal.Deposit += txDeposit.Deposit
+            
+            if (proposal.Deposit >= ActiveProcedure.MinDeposit) then  
+              // MinDeposit is reached, vote opens
+              
+              proposal.VotingStartBlock = CurrentBlock
+              
+              create  votersList,
+                      initVotingPowerList,
+                      minusesList,
+                      initProcedureNumber,
+                      initTotalVotingPower  from proposalID
+              
+              snapshot(ActiveProcedure.ProcedureNumber) // Save current procedure number in InitProcedureNumber
+              snapshot(TotalVotingPower)  // Save total voting power in InitTotalVotingPower
+              snapshot(ValidatorVotingPower)  // Save validators' voting power in InitVotingPowerList
+              
+              ProposalProcessingQueueEnd++  // ProposalProcessingQueue will have a new element
+              ProposalProcessingQueue[ProposalProcessingQueueEnd] = txDeposit.ProposalID  
+```
+
+Finally, if the proposal is accepted or `MinDeposit` was not reached before the end of the `MaximumDepositPeriod`, then Atom holders can send `TxClaimDeposit` transaction to claim their deposits.
+
+```Go
+  type TxClaimDeposit struct {
```

Comment:
`TxGovClaimDeposit`? - maybe these transaction names should reference the fact that they apply to goverenace throughout?

Reply:
Yes, done